static/skin/taskbar.js

```diff
@@ -5,24 +5,38 @@ function htmlDecode(input) {
 
 (function ($) {
     const root = $( `link[type='root']` ).attr("href");
+    const bookName = window.location.pathname.split(`${root}/`)[1].split('/')[0];
 
     $( "#kiwixsearchbox" ).autocomplete({
 
-        source: `${root}/suggest?content=${window.location.pathname.split(`${root}/`)[1].split('/')[0]}`,
+        source: `${root}/suggest?content=${bookName}`,
         dataType: "json",
         cache: false,
 
         response: function( event, ui ) {
 
               for(const item of ui.content) {
                   item.label = htmlDecode(item.label);
-                  item.value = htmlDecode(item.value);
+                  item.value = encodeURI(htmlDecode(item.value));
+
+                  if (item.label === `containing '${item.value}'...`) {
+                      item.value = `${root}/search?content=${bookName}&pattern=${item.value}+`;
+                  }
+                  else {
+                      item.value = `${root}/${bookName}/${item.value}`;
+                  }
               }
         },
 
+        focus: function(event, ui) {
+            $( "#kiwixsearchbox" ).val(ui.item.label);
+            return false;
+        },
+
         select: function(event, ui) {
-        $( "#kiwixsearchbox" ).val(ui.item.value);
-        $( "#kiwixsearchform" ).submit();
+            window.location.href = ui.item.value;
+            $( "#kiwixsearchbox" ).val(ui.item.label);
+            return false;
```

Comment:
We already have the search path "computed" by the form (https://github.com/kiwix/kiwix-lib/blob/master/static/templates/taskbar_part.html#L5-L9).
It would be good to avoid compute it in another place (for now it is compute in `response` function, but as said in another commit, it should be handled here).

If the item is a existing entry path, we must directly change the `window.location.href` and if this is a real search, let's use the form to do it.
(If we need to change the search url, or add a query parameter, we will have to do it in only one place).

Reply:
@mgautierfr if we are submitting the from then what is the need of creating another response element named `pattern(or term)` as for that case `ui.item.value` stores the exact same thing

Reply:
we should then just create `path` and `kind`